File: app/src/main/java/com/charlie/sawl/homepage/Homepage.java
package com.charlie.sawl.homepage;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;
import android.support.v4.view.PagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AlertDialog;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import com.charlie.sawl.R;

import java.util.Objects;

public class Homepage extends Fragment {

    private final String[] pageTitle = {"공지사항", "가정통신문"};

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setHasOptionsMenu(true);
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_tab, container, false);

        TabLayout tabLayout = rootView.findViewById(R.id.mTabLayout);
        int numOfPages = 2;
        for (int i = 0; i < numOfPages; i++) {
            tabLayout.addTab(tabLayout.newTab().setText(pageTitle[i]));
        }
        tabLayout.setTabGravity(TabLayout.GRAVITY_FILL);
        final ViewPager viewPager = rootView.findViewById(R.id.mViewPager);
        final PagerAdapter adapter = new HomepageAdapter(getFragmentManager(), numOfPages);

        viewPager.setAdapter(adapter);
        viewPager.addOnPageChangeListener(new TabLayout.TabLayoutOnPageChangeListener(tabLayout));
        viewPager.setOffscreenPageLimit(2);
        tabLayout.addOnTabSelectedListener(onTabSelectedListener(viewPager));

        return rootView;
    }

    @Override
    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
        inflater.inflate(R.menu.menu_homepage, menu);
        super.onCreateOptionsMenu(menu, inflater);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.information:
                AlertDialog.Builder builder = new AlertDialog.Builder(Objects.requireNonNull(getActivity()));
                builder.setTitle("공지사항 본문이 보이지 않아요!");
                builder.setMessage("본문이 순수한 텍스트로 되어 있을 경우에만 확인이 가능합니다. 사진, 표 등 뿐만 아니라 특수효과(밑줄 등)가 적용된 텍스트 역시 확인할 수 없습니다.");
                builder.setPositiveButton(android.R.string.ok, null);
                builder.show();
                break;
        }
        return true;
    }

    private TabLayout.OnTabSelectedListener onTabSelectedListener(final ViewPager pager) {
        return new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {
                pager.setCurrentItem(tab.getPosition());
            }

            @Override
            public void onTabUnselected(TabLayout.Tab tab) {}

            @Override
            public void onTabReselected(TabLayout.Tab tab) {}
        };
    }

    public class HomepageAdapter extends FragmentStatePagerAdapter {

        int numOfTabs;

        HomepageAdapter(FragmentManager fm, int numOfTabs) {
            super(fm);
            this.numOfTabs = numOfTabs;
        }

        @Override
        public Fragment getItem(int position) {
            switch(position) {
                case 0:
                    return new Notice();
                case 1:
                    return new Parents();
            }

            return null;
        }

        @Override
        public int getCount() {
            return numOfTabs;
        }
    }
}

File: app/src/main/kotlin/com/charlie/sawl/homepage/Homepage.kt
package com.charlie.sawl.homepage

import android.os.Bundle
import com.google.android.material.tabs.TabLayout
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import androidx.viewpager.widget.ViewPager
import androidx.appcompat.app.AlertDialog
import android.view.LayoutInflater
import android.view.Menu
import android.view.MenuInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import com.charlie.sawl.R

class Homepage : Fragment() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setHasOptionsMenu(true)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val rootView = inflater.inflate(R.layout.fragment_tab, container, false)

        val tabLayout = rootView.findViewById<TabLayout>(R.id.mTabLayout)
        tabLayout.addTab(tabLayout.newTab().setText(R.string.notice))
        tabLayout.addTab(tabLayout.newTab().setText(R.string.parent))
        tabLayout.tabGravity = TabLayout.GRAVITY_FILL
        val viewPager = rootView.findViewById<ViewPager>(R.id.mViewPager)
        val adapter = HomepageAdapter(activity!!.supportFragmentManager, 2)

        viewPager.adapter = adapter
        viewPager.addOnPageChangeListener(TabLayout.TabLayoutOnPageChangeListener(tabLayout))
        viewPager.offscreenPageLimit = 2
        tabLayout.addOnTabSelectedListener(onTabSelectedListener(viewPager))

        return rootView
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        inflater.inflate(R.menu.menu_homepage_fragment, menu)
        super.onCreateOptionsMenu(menu, inflater)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.information -> {
                val builder = AlertDialog.Builder(activity!!)
                builder.setTitle("공지사항 본문이 보이지 않아요!")
                builder.setMessage("본문이 순수한 텍스트로 되어 있을 경우에만 확인이 가능합니다. 사진, 표 등 뿐만 아니라 특수효과(밑줄 등)가 적용된 텍스트는 확인이 어려울 수 있습니다.")
                builder.setPositiveButton(android.R.string.ok, null)
                builder.show()
            }
        }
        return true
    }

    private fun onTabSelectedListener(pager: ViewPager): TabLayout.OnTabSelectedListener {
        return object : TabLayout.OnTabSelectedListener {
            override fun onTabSelected(tab: TabLayout.Tab) {
                pager.currentItem = tab.position
            }

            override fun onTabUnselected(tab: TabLayout.Tab) {}

            override fun onTabReselected(tab: TabLayout.Tab) {}
        }
    }

    inner class HomepageAdapter internal constructor(fm: FragmentManager, private var numOfTabs: Int) : FragmentStatePagerAdapter(fm) {

        override fun getItem(position: Int): Fragment {
            when (position) {
                0 -> return Notice()
                1 -> return Parent()
            }

            return Notice()
        }

        override fun getCount(): Int {
            return numOfTabs
        }
    }
}